### PR TITLE
Kaldi STT

### PR DESCRIPTION
**Pull request by Teagan42**
Finish https://github.com/MycroftAI/mycroft-core/pull/440

Can someone with Kaldi please verify - I had to wipe my server the other day and haven't stood kaldi back up.

### CODE REVIEW

File: mycroft/stt/__init__.py [hunk 1]
```diff
@@ -106,12 +108,31 @@ def execute(self, audio, language=None):
...
 
 
+class KaldiSTT(STT):
+    def __init__(self):
+        super(KaldiSTT, self).__init__()
+
+    def execute(self, audio, language=None):
+        language = language or self.lang
+        config = ConfigurationManager.get().get("stt", {})
+        response = requests.post(self.config.get("uri"), data=audio.get_wav_data())
```
Comment (augustnmonteiro) [inline mycroft/stt/__init__.py]: Can you split this line, is more than 80 columns it brokes pep8, thank you

**Comment by forslund** [comment]
I'm currently building Kaldi to give this a try. but it seems to take forever! If someone with an existing install would like to test this my struggling CPU would be happy =)

File: mycroft/stt/__init__.py (hunk 1 above)
Comment (forslund) [inline mycroft/stt/__init__.py]: also: the module requests need to be imported at the top of the file. You can do

```python
from requests import post
```

And then shorten this line by just making it
`response = post(self.config.get("uri"), data=audio.get_wav_data())`

**Comment by augustnmonteiro** [comment]
@forslund I mine just finished to compile, I'm downloading the models to run, but yeah, takes a lot of time

File: mycroft/stt/__init__.py [hunk 2]
```diff
@@ -106,12 +108,31 @@ def execute(self, audio, language=None):
...
+        return get_response(response)
```
Comment (forslund) [inline mycroft/stt/__init__.py]: this should be
```python
        return self.get_response(response)
```

File: mycroft/stt/__init__.py [hunk 3]
```diff
@@ -106,12 +108,31 @@ def execute(self, audio, language=None):
...
+
+    def get_response(response):
```
Comment (forslund) [inline mycroft/stt/__init__.py]: this should be 
```python
    def get_response(self, response):
```

**Comment by Teagan42** [comment]
Sorry about that - wrote it real quick and didn't have a mycroft instance to test it against, in the middle of rebuilding some servers at home. Please check the latest commit

File: mycroft/stt/__init__.py (hunk 1 above)
Comment (Teagan42) [inline mycroft/stt/__init__.py]: yup, yup and yup.

File: mycroft/stt/__init__.py (hunk 2 above)
Comment (Teagan42) [inline mycroft/stt/__init__.py]: Context switching between python for some side projects and Java for work - my bad. addressed.

File: mycroft/stt/__init__.py (hunk 3 above)
Comment (Teagan42) [inline mycroft/stt/__init__.py]: addressed.

File: mycroft/stt/__init__.py [hunk 4]
```diff
@@ -106,12 +110,31 @@ def execute(self, audio, language=None):
         return self.api.stt(audio.get_flac_data(), self.lang, 1)[0]
 
 
+class KaldiSTT(STT):
+    def __init__(self):
+        super(KaldiSTT, self).__init__()
+
+    def execute(self, audio, language=None):
+        language = language or self.lang
+        config = ConfigurationManager.get().get("stt", {})
```
Comment (forslund) [inline mycroft/stt/__init__.py]: This config isn't actually used and can be removed

**Comment by feitingen** [comment]
Great, it works for me with my kaldi setup!

May I suggest altering the comments in `mycroft/configuration/mycroft.conf`?
Specifically around 156:

`  // Speech to Text parameters`
`  // Override: REMOTE`
`  "stt": {`
`    // Engine.  Options: "mycroft", "google", "wit", "ibm"`
`    "module": "mycroft"`
`  },`


to something like
`  // Speech to Text parameters`
`  // Override: REMOTE`
`  "stt": {`
`    // Engine.  Options: "mycroft", "google", "wit", "ibm", "kaldi"`
`    "module": "mycroft"`
`    // URI required for kaldi`
`    // "uri": "http://localhost:8080/client/dynamic/recognize"`
`  },`

**Comment by forslund** [comment]
Looking at the base class the config should really be something like
```json
"stt": {
  "module": "kaldi",
  "kaldi": {
    "uri": "http://localhost"
  }
}
```

using this setup, `self.config` (which gets mapped to stt['kaldi']) can be used instead of the local variable that's created.

This is the format used by the other STT backends that need extra parameters.

**Comment by Teagan42** [comment]
@forslund addressed. You're right, I should have looked at the base class more carefully

**Comment by forslund** [comment]
Nice! Thank you!

**Comment by feitingen** [comment]
I checked out again, and still works nicely on my desktop.
Thank you!

**Comment by Teagan42** [comment]
Can we get this merged up?

**Comment by augustnmonteiro** [comment]
@Teagan42 thank you!!!

**Pull request merged**